File: javascript/claude/p11_nr_ver3.js
"use strict";

const ID_SEQUENCE = [10, 20, 30, 40, 50];

function getValueFromArray(arr, size, index) {
    if (!Array.isArray(arr) || size <= 0 ||
        !Number.isInteger(index) || index < 0 || index >= size) {
        return -1;
    }
    return arr[index];
}

function main() {
    const chunks = [];

    process.stdin.on("data",  chunk  => chunks.push(chunk));
    process.stdin.on("end",   ()     => {
        const raw = chunks.join("").trim();

        if (!raw) process.exit(-1);

        // Number("3.5") is not an integer; String round-trip rejects "3e2" etc.
        const parsed = Number(raw);
        if (!Number.isInteger(parsed) || String(parsed) !== raw) {
            process.exit(-1);
        }

        const result = getValueFromArray(ID_SEQUENCE, ID_SEQUENCE.length, parsed);
        process.exit(result);
    });
}

main();

File: python/claude/p11_nr_ver3.py
import sys

id_sequence = [10, 20, 30, 40, 50]

def get_value_from_array(arr: list[int], size: int, index: int) -> int:
    if not arr or size <= 0 or not isinstance(index, int) or index < 0 or index >= size:
        return -1
    return arr[index]

def main() -> None:
    raw = sys.stdin.readline().strip()

    if not raw:
        sys.exit(-1)

    try:
        # int() rejects floats ("3.5"), leading whitespace is already stripped
        index = int(raw)
    except ValueError:
        sys.exit(-1)

    result = get_value_from_array(id_sequence, len(id_sequence), index)
    sys.exit(result)

if __name__ == "__main__":
    main()